emote/algos/genrl/proxies.py

```diff
@@ -0,0 +1,46 @@
+from typing import Dict, Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from emote.memory.memory import TableMemoryProxy
+from emote.memory.table import Table
+from emote.typing import AgentId, DictObservation, DictResponse
+
+
+class MemoryProxyWithEncoder(TableMemoryProxy):
+    def __init__(
+        self,
+        table: Table,
+        encoder: nn.Module,
+        minimum_length_threshold: Optional[int] = None,
+        use_terminal: bool = False,
+        input_key: str = "obs",
+    ):
+        super().__init__(table, minimum_length_threshold, use_terminal)
+        self.encoder = encoder
+        self._input_key = input_key
```

Comment:
Curve-ball: If we wanted to do a hidden-state style memory network (GRU, LSTM, ...) could this support it, or what would need to change?

Reply:
We construct a new MDP with a different action space, which we call it latent action space. The proxy first maps actions into the latent action space and then stores the encoded actions instead of the original ones. As long as there is a 1-1 mapping between the original actions and the latent actions, there is no added complexity using genrl. This is the case (1-1 mapping) when using only action data for the mapping. It is also fine when conditioning on observation since we only get the proprioceptive data (we do not deal with partial observations).